YAML file .xslx selection support

korgano opened the pull request:
Addresses issue #12. Have successfully proven out generation, although it requires updating some dependencies.

Also adds basic documentation on the YAML file options, including information about adding eras to the spreadsheet.

korgano committed:
Update main.ts - Support .config.yaml specifying spreadsheet

Handles reading and logging spreadsheet called out in .config.yaml, falls back to generic settings in data-source.config.yaml

korgano committed:
Create interfaces.ts - support .config.yaml specifying spreadsheet

korgano committed:
Create config-utils.ts - support .config.yaml specifying spreadsheet

korgano committed:
Create config-merge.ts - support .config.yaml specifying spreadsheet

korgano committed:
Create map-gen-cusomization.md

Initial map customization documentation

korgano committed:
Rename map-gen-cusomization.md to map-gen-customization.md

Fixed typo in file name.

korgano committed:
Update README.md - map customization document link

Added link for basic map customization documentation.

aquachris reviewed (changes requested):
I love the customization doc file, thank you for that!

As for the config merge code, what problem are we trying to solve here? I don't quite understand what it's good for.

Can we split this PR into two, in order to merge the documentation improvements?

korgano commented:
> As for the config merge code, what problem are we trying to solve here? I don't quite understand what it's good for.

It's **really** good for people who want to do multiple variants of the same era and just general data organization.

In my case, generating maps for different BattleTech AUs that are affecting the Clan invasion era.

It's easier and cleaner to make a specific variant SUCKit spreadsheet and point the config file to use it as a source, than it is to add all the variants to one spreadsheet, make sure they're all labeled correctly, and use the era index to organize them. And it also allows me to fallback to the normal SUCKit config if I don't declare anything, which is great when I want a canon map.

So, for example, I want to make maps for the Coyote Games AU, where the Ghost Bears are invading the FedCom, the Coyotes are invading the FedCom/FRR, the Star Adders are invading the FRR and Draconis Combine, and the Jade Falcons are in the Combine. But I also want to make maps for Ice Hellion: Offside, where the Ice Hellions are invading Sapporo during the canon Clan Invasion.

Those are two entirely different maps, and it'd be a mess to have them in the same spreadsheet.

aquachris commented:
Ok, so if I understand you correctly (please tell me if I'm wrong), the issue is using different xlsx files for different generator configs, with the official SUCKit as a fallback, right? If so, I think there's probably a simpler implementation, I can take a crack at it.

korgano commented:
> Ok, so if I understand you correctly (please tell me if I'm wrong), the issue is using different xlsx files for different generator configs, with the official SUCKit as a fallback, right? If so, I think there's probably a simpler implementation, I can take a crack at it.

Yes, that is the intended use case.

I'll see about making a separate PR for the documentation, unless you want me to hold off until the code is refactored.

korgano committed:
Improved YAML localFileConfig + tracing faction/affiliation

Simpler localFileConfig reading from YAML.

Working towards fixing bugs regarding factions with letters + numbers in string.

korgano committed:
Fixed Letter/Digit Factions on pre-2026 Excel files

This code is known to work on pre-2026 versions of the SUCKit spreadsheet.

2026 versions have multiple issues:
- lack of proper system dot generation
- columns with custom factions seem to cause chaotic map generation

aquachris reviewed (approved):
I'm not going to merge this into main, as there are too many changes, and some I do not want to make (such as dropping XLSX support).
However, the contributions are very valuable, and I'm going to keep a separate branch and cherry pick specific changes and fixes.
